thrift/lib/py/protocol/TJSONProtocol.py

```diff
@@ -208,7 +219,30 @@ def readJSONSyntaxChar(self, character):
             raise TProtocolException(TProtocolException.INVALID_DATA,
                                      "Unexpected character: %s" % current)
 
+    def _isHighSurrogate(self, codeunit):
+        return codeunit >= 0xd800 and codeunit <= 0xdbff
+
+    def _isLowSurrogate(self, codeunit):
+        return codeunit >= 0xdc00 and codeunit <= 0xdfff
+
+    def _toChar(self, high, low=None):
+        if not low:
+            if sys.version_info[0] == 2:
+                return ("\\u%04x" % high).decode('unicode-escape') \
+                                         .encode('utf-8')
+            else:
+                return chr(high)
+        else:
+            codepoint = (1 << 16) + ((high & 0x3ff) << 10)
+            codepoint += low & 0x3ff
+            if sys.version_info[0] == 2:
+                s = "\\U%08x" % codepoint
+                return s.decode('unicode-escape').encode('utf-8')
+            else:
+                return chr(codepoint)
+
     def readJSONString(self, skipContext):
+        highSurrogate = None
```

Comment:
This seems to be used as a boolean flag even though we're assigning integers to it. Maybe we replace this with a boolean for maintainability? `previousIsHighSurrogate`?

Reply:
We spoke about this offline. Decided to let it be as its copy pasted from the original Thrift so we probably want to keep that part of the code as similar as possible for any future copies.